server/src/main/java/org/tctalent/server/service/db/impl/SavedSearchServiceImpl.java

```diff
@@ -137,781 +136,777 @@
+    if (user == null) {
+      candidates = doSearchCandidates(request);
+    } else {
+      //Update default search
+      SavedSearch defaultSavedSearch =
+          getDefaultSavedSearch();
+      Long savedSearchId = defaultSavedSearch.getId();
+      UpdateSavedSearchRequest updateRequest = new UpdateSavedSearchRequest();
+      updateRequest.setSearchCandidateRequest(request);
+      //Set other fields - no changes there
+      updateRequest.setName(defaultSavedSearch.getName());
+      updateRequest.setDefaultSearch(defaultSavedSearch.getDefaultSearch());
+      updateRequest.setFixed(defaultSavedSearch.getFixed());
+      updateRequest.setReviewable(defaultSavedSearch.getReviewable());
+      updateRequest.setSavedSearchType(defaultSavedSearch.getSavedSearchType());
+      updateRequest.setSavedSearchSubtype(defaultSavedSearch.getSavedSearchSubtype());
+      //todo Need special method which only updates search part. Then don't need the above "no changes there" stuff
+      updateSavedSearch(savedSearchId, updateRequest);
+
+      //Do the search
+      candidates = doSearchCandidates(request);
+
+      //Add in any selections
+      markUserSelectedCandidates(savedSearchId, candidates);
     }
 
-    @Override
-    public SearchCandidateRequest loadSavedSearch(long id) {
-        SavedSearch savedSearch = this.savedSearchRepository.findByIdLoadSearchJoins(id)
-                .orElseThrow(() -> new NoSuchObjectException(SavedSearch.class, id));
-        savedSearch.parseType();
-        return convertToSearchCandidateRequest(savedSearch);
+    return candidates;
+  }
+
+  /**
+   * Mark the Candidate objects with any context associated with the selection list of the saved search. This means that context fields (ie
+   * ContextNote) associated with the saved search will be returned through the DtoBuilder if present.
+   */
+  @Override
+  public void setCandidateContext(long savedSearchId, Iterable<Candidate> candidates) {
+    User user = userService.getLoggedInUser();
+    SavedList selectionList = null;
+    if (user != null) {
+      selectionList = getSelectionList(savedSearchId, user.getId());
+    }
+    if (selectionList != null) {
+      //Only set context of selected candidates
+      Set<Candidate> selectedCandidates = selectionList.getCandidates();
+      //Loop through candidates we are considering
+      for (Candidate candidate : candidates) {
+        //Only selected candidates
+        if (selectedCandidates.contains(candidate)) {
+          candidate.setContextSavedListId(selectionList.getId());
+        }
+      }
+    }
+  }
+
+  @Override
+  public SearchCandidateRequest loadSavedSearch(long id) {
+    SavedSearch savedSearch = this.savedSearchRepository.findByIdLoadSearchJoins(id)
+        .orElseThrow(() -> new NoSuchObjectException(SavedSearch.class, id));
+    savedSearch.parseType();
+    return convertToSearchCandidateRequest(savedSearch);
+  }
+
+  @Override
+  public SavedSearch getSavedSearch(long id) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.